src/components/organisms/PaymentModal/PaymentForm.tsx

```diff
@@ -61,32 +64,69 @@ const PaymentForm: React.FunctionComponent<PropsType> = ({
     }
   }, [event.id, venueId, ticketPrice, user, selectedTab]);
 
-  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+  const handleSubmit = async (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>,
+    setIsLoading: (isLoading: boolean) => void,
+    operations: (stripe: Stripe, elements: StripeElements) => Promise<void>
+  ) => {
     if (!user) return;
-    setIsFormBeingSubmitted(true);
+    setIsLoading(true);
     e.preventDefault();
     if (!stripe || !elements) {
       setErrorMessage("Oops something wrong happened");
+      setIsLoading(false);
       return;
     }
-    const result = await stripe.confirmCardPayment(clientSecret || "", {
+    await operations(stripe, elements);
+    setIsLoading(false);
+  };
+
+  const handlePayment = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) =>
+    handleSubmit(e, setIsPaymentProceeding, pay);
+  const handleSaveCard = (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) =>
+    handleSubmit(e, setIsCardBeingSaved, saveCardAndPay);
+
+  const pay = async (stripe: Stripe, elements: StripeElements) => {
+    if (!(user?.email && clientSecret)) {
+      setErrorMessage("Oops something wrong happened");
+      return;
+    }
+    const result = await stripe?.confirmCardPayment(clientSecret || "", {
```

Comment:
clientSecret should exist here. Also can we now get rid of `stripe?`?. If not, add if(!stripe) to check it exists